docs/tutorials/data_management.md

```diff
@@ -48,204 +48,184 @@ If you check in the generated data, it is recommended that you configure a job i
+The `--keys-for-bin` argument tells `icu4x-datagen` to analyze the binary and only include keys that are used by its code. This significantly reduces the blob's file size, to 54KB, and our program still works. Quite the improvement!
 
-Same as `BlobDataProvider`, this also a buffer provider, so you will need to activate `icu`'s `serde` Cargo feature and use the `with_buffer_provider` constructors.
+But there is more to optimize. You might have noticed this in the output of the `icu4x-datagen` invocation, which lists 21 keys, including clearly irrelevant ones like `datetime/ethopic/datesymbols@1`. Remember how we had to convert our `DateTime<Gregorian>` into a `DateTime<AnyCalendar>` in order to use the `DateTimeFormatter`? Turns out, as `DateTimeFormatter` contains logic for many different calendars, datagen includes data for all of these as well.
 
-```console
-$ cargo add icu --features serde
-$ cargo add icu_provider --features deserialize_json
-$ cargo add icu_provider_fs
-```
+We can instead use `TypedDateTimeFormatter<Gregorian>`, which only supports formatting `DateTime<Gregorian>`s:
 
 ```rust,no_run
 use icu::locid::{locale, Locale};
-use icu::calendar::DateTime;
+use icu::calendar::{DateTime, Gregorian};
 use icu::datetime::{TypedDateTimeFormatter, options::length};
-use icu_provider_fs::FsDataProvider;
+use icu_provider_blob::BlobDataProvider;
 
 const LOCALE: Locale = locale!("ja");
 
 fn main() {
-    let buffer_provider = FsDataProvider::try_new("my-data-dir")
-           .expect("Failed to initialize Data Provider");
+    let blob = std::fs::read("my-data-blob").expect("Failed to read file");
+    let buffer_provider = 
+        BlobDataProvider::try_new_from_blob(blob.into_boxed_slice())
+            .expect("blob should be valid");
 
     let options = length::Bag::from_date_time_style(length::Date::Long, length::Time::Medium);
 
-    let dtf = TypedDateTimeFormatter::try_new_with_buffer_provider(&buffer_provider, &LOCALE.into(), options.into())
-        .expect("Failed to initialize DateTimeFormatter");
+    let dtf = TypedDateTimeFormatter::<Gregorian>::try_new_with_buffer_provider(&buffer_provider, &LOCALE.into(), options.into())
+        .expect("blob should contain required data");
 
     let date = DateTime::try_new_gregorian_datetime(2020, 10, 14, 13, 21, 28)
-        .expect("Failed to create a datetime.");
+        .expect("datetime should be valid");
 
     let formatted_date = dtf.format(&date);
 
     println!("📅: {}", formatted_date);
 }
 ```
 
+This has two advantages: it reduces our code size, as `DateTimeFormatter` includes much more functionality than `TypedDateTimeFormatter<Gregorian>`, and it reduces our data size, as `--keys-for-bin` can now determine that we need even fewer keys. The data size improvement could have also been achieved by manually listing the data keys we think we'll need (using the `--keys` flag), but we risk a runtime error if we're wrong.
```

Comment:
Nit: If listing out the advantages, please also list the disadvantages

```suggestion
This has two advantages: it reduces our code size, as `DateTimeFormatter` includes much more functionality than `TypedDateTimeFormatter<Gregorian>`, and it reduces our data size, as `--keys-for-bin` can now determine that we need even fewer keys. However, this approach causes regions and users that prefer a non-Gregorian calendar to see their dates rendered in Gregorian, so it should be taken only after sufficient research on application requirements.

Aside: The data size improvement could have also been achieved by manually listing the data keys we think we'll need (using the `--keys` flag), but we risk a runtime error if we're wrong.
```

Feel free to move the aside further down in the section.

Reply:
I think this is information overload. This toy application also renders only `ja` output, and we don't have a disclaimer that this will be wrong for non-Japanese users.

Reply:
I would also be okay if you changed the example to force a non-Gregorian calendar like Japanese.

My worry, which I believe has a high probability of coming to pass, is that developers reading this don't know what a non-Gregorian calendar is and will think "hey, that's cool! I save data by doing this thing so I should definitely do this thing" without understanding the impact. When you select a single language, the impact on i18n is more understandable to the average developer than when you select Gregorian-only.

Reply:
I can do that later. We're only merging this after the release anyway and still have to update all the versions.